src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -574,17 +568,13 @@ public ActiveDirectoryGroupDetails call() throws NamingException {
                             }
                             LOGGER.log(Level.WARNING, "Exhausted all configured domains and could not authenticate against any");
                             throw new UserMayOrMayNotExistException(groupname);
-                        }
                     });
-        } catch (UncheckedExecutionException e) {
+        } catch (Exception e) {
```

Comment:
This is now catching `UserMayOrMayNotExistException` and wrapping it in `CacheAuthenticationException` which is a `RuntimeException`, causing HTTP 403 error instead of just being handled as a `UsernameNotFoundException` by the caller.